### PR TITLE
MultiThresholdObjects: Create BoolArray instead of UInt8Array

### PR DESCRIPTION
**Pull request by imikejackson**
Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

### CODE REVIEW

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]
All the uses of the mask array need to updated as well.

**Review by joeykleingers — APPROVED** [review]
Once Jared's changes get put in, this looks good.

**Comment by imikejackson** [comment]
Do we know what filters have mask arrays?

**Comment by joeykleingers** [comment]
Just from a cursory glance, here are all the filters that have masks with types that need to be changed to boolean:

FindSurfaceFeatures
MultiThresholdObjects
RobustAutomaticThreshold
ScalarSegmentFeaturesFilter

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/MultiThresholdObjects.cpp [hunk 1]
```diff
@@ -244,7 +245,10 @@ void ThresholdValue(std::shared_ptr<ArrayThreshold>& comparisonValue, DataStruct
   {
     if(inverse)
     {
-      std::for_each(tempResultVector.begin(), tempResultVector.end(), [](uint8_t& n) { (n == 0 ? n = 1 : n = 0); });
+      for(size_t i = 0; i < tempResultVector.size(); i++)
+      {
+        tempResultVector[i] = !tempResultVector[i];
+      }
```
Comment (joeykleingers) [inline src/Plugins/ComplexCore/src/ComplexCore/Filters/MultiThresholdObjects.cpp:251]:
```suggestion
      tempResultVector.flip();
```

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/MultiThresholdObjects.cpp [hunk 2]
```diff
@@ -306,7 +310,10 @@ void ThresholdSet(std::shared_ptr<ArrayThresholdSet>& inputComparisonSet, DataSt
...
```
Comment (joeykleingers) [inline src/Plugins/ComplexCore/src/ComplexCore/Filters/MultiThresholdObjects.cpp:316]:
```suggestion
      tempResultVector.flip();
```

**Review by joeykleingers — CHANGES_REQUESTED** [review]

**Review by joeykleingers — DISMISSED** [review]

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/MultiThresholdObjects.cpp [hunk 3]
```diff
@@ -185,23 +185,24 @@ class ThresholdFilterHelper
...
+void InsertThreshold(int64_t numItems, BoolArray& currentArray, complex::IArrayThreshold::UnionOperator unionOperator, std::vector<bool>& newArrayPtr, bool inverse)
 {
+
   for(int64_t i = 0; i < numItems; i++)
   {
     // invert the current comparison if necessary
     if(inverse)
     {
-      (newArrayPtr[i] == 0 ? newArrayPtr[i] = 1 : newArrayPtr[i] = 0);
+      (!newArrayPtr[i] ? newArrayPtr[i] = true : newArrayPtr[i] = false);
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/src/ComplexCore/Filters/MultiThresholdObjects.cpp:196]:
```suggestion
      newArrayPtr[i] = !newArrayPtr[i];
```

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/MultiThresholdObjects.cpp [hunk 4]
```diff
@@ -185,23 +185,24 @@ class ThresholdFilterHelper
...
     }
 
     if(complex::IArrayThreshold::UnionOperator::Or == unionOperator)
     {
-      (currentArray[i] != 0 || newArrayPtr[i] != 0) ? currentArray[i] = 1 : currentArray[i] = 0;
+      (currentArray[i] || newArrayPtr[i]) ? currentArray[i] = true : currentArray[i] = false;
```
Comment (JDuffeyBQ) [inline src/Plugins/ComplexCore/src/ComplexCore/Filters/MultiThresholdObjects.cpp:201]:
```suggestion
      currentArray[i] = currentArray[i] || newArrayPtr[i];
```

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]

**Review by JDuffeyBQ — APPROVED** [review]

**Pull request merged**